src/BaseComponent.php

```diff
@@ -91,36 +90,25 @@ protected function loadConfig(): void
 
     protected function loadInputState(): void
     {
-        $inputStateFile = $this->getDataDir() . '/in/state.json';
-        if (file_exists($inputStateFile)) {
-            $jsonContents = (string) file_get_contents($inputStateFile);
-            $jsonEncoder = new JsonEncoder();
-            $this->inputState = $jsonEncoder->decode($jsonContents, JsonEncoder::FORMAT);
-        } else {
+        try {
+            $this->inputState = (new JsonFileHelper())->read($this->getDataDir() . '/in/state.json');
+        } catch (FileNotFoundException $exception) {
             $this->inputState = [];
         }
     }
 
     protected function writeOutputStateToFile(array $state): void
     {
-        $outDataDir = $this->getDataDir() . '/out';
-        if (!is_dir($outDataDir)) {
-            mkdir($outDataDir);
-        }
-
-        $outputStateFile = $outDataDir . '/state.json';
-        $jsonEncode = new JsonEncoder();
-        file_put_contents(
-            $outputStateFile,
-            $jsonEncode->encode($state, JsonEncoder::FORMAT, ['json_encode_options' => JSON_PRETTY_PRINT])
+        (new JsonFileHelper())->write(
+            $this->getDataDir() . '/out/state.json',
+            $state,
+            ['json_encode_options' => JSON_PRETTY_PRINT]
```

Comment:
nehodilo by se to jako default? Jak často zapisujeme json neformátovaný? Osobně bych radši viděl true/false parametr, protože posílat celý kontext mi přijde potenciálně problematické.

Reply:
Nekde jsem neformatovany json uz videl - myslim, ze v teste db-ex-common/wr-bigquery. Nicmene zrovna na tech mistech mi prijde lepsi to mit naformatovane, at se v tom clovek dokaze jednoduze zorientovat.

Mozna nemusime posilat cely kontext, ale primo `json_encode_options `, tim budes schopny ten json dodatecne nastavit. Nenapada me ale pripad, kdy pouzivame neco jine nez `JSON_PRETTY_PRINT`.

Jestli vas taky nic nenapada tak to muzeme resit `true`/`false` parametrem.

Reply:
spis je mozna otazka - potrebujem tam to nastaveni vubec na neco? jestli jo, tak bych to dal jako default a bool, nic jinyho jsme nikdy nepotrebovali (a kdyby to nekdy nekdo potrevoval, tak se to da vzdycky napsal low-level)

Reply:
osobne bych to teda vubec nedaval nastavitelny, at je to co nejjednodussi

Reply:
Pridal jsem tam ten bool, a defaultne to nastavil na true. Kdyz to budes chtit vypsat na jeden radek, musis tam dopsat false.

Reply:
btw, co kdyby to byly staticky metody? helper zadnej vnitrni stav nema a mit nebude a 
JsonFileHelper::write(...) je jednodussi nez `new JsonFileHelper())->write(...)`